### ISSUE TITLE
Plugin switches IDE to a wrong theme

### ISSUE THREAD

**Issue by kuznecc** [weisJ/auto-dark-mode#19]
After the last update, I noticed the strange behaviour of a plugin. 
It looks like the plugin sets the IDE theme vice versa to the OS theme.
Here is a video below. 
1. I starting the IDEA
2. plugin switches the IDE theme to match with OS
3. then I switching the OS theme and the plugin ignores this change
4. every consequent change of the OS theme triggers the plugin to switch the IDE theme but to the opposite one. (OS=dark -> IDE=light)

![Untitled](https://user-images.githubusercontent.com/2734183/90103116-aabc3180-dd4a-11ea-8ebb-730cd9ec10bc.gif)

Here are the plugin settings
<img width="286" alt="Screenshot 2020-08-13 at 09 55 19" src="https://user-images.githubusercontent.com/2734183/90103437-2b7b2d80-dd4b-11ea-8a9a-cd1c179280fb.png">

**Comment by weisJ** [comment]
I’ll need some information from you:
- IntelliJ version
- macOS version

Is this the first time you installed the plugin for the specific IntelliJ installation? If yes you will have to restart IntelliJ at least once.
In the meantime you could switch your dark/light settings to accommodate the issue.

**Comment by weisJ** [comment]
Do you experience the same issue if you change to dark/light mode in the official macOS settings?

**Comment by kuznecc** [comment]
Idea : 2020.2
MacOs : 10.15.6
I using your amazing plugin for a long time. 
The problem appears after the latest update, that I've made yesterday.
To record the video I restarted the IDEA a bunch of times. 
Actually the bug can be reproduced right after the IDE startup. During a startup, the plugin sets the correct theme, but after I starting to switch them it doesn't do anything for the first 'change' and for all consequent changes starts to work but vise versa.

I tried to switch the macOS theme all the times with 'NightOwl' app. And seems this is a culprit. 
When I switch the OS theme with the OS system properties the plugin is working flawlessly.
It's strange, bcz previously I didn't notice any issues. 
Sorry for bothering :(

**Comment by weisJ** [comment]
Then I think I might know what is causing the issue. I’ll have to do some research on how NightOwl does it’s theme switching and see whether I can do something to support it.

**Comment by dobrovolsky** [comment]
The same issue with another approach, using hammerspoon.
Maybe it will be easier to understand what is going on.
https://github.com/dobrovolsky/dotfiles/blob/master/config/.hammerspoon/darkmode.lua#L45

**Comment by weisJ** [comment]
> The same issue with another approach, using hammerspoon.
> Maybe it will be easier to understand what is going on.
> https://github.com/dobrovolsky/dotfiles/blob/master/config/.hammerspoon/darkmode.lua#L45

This is just calling some javascript, so sadly not really helpful. I have already gotten an answer by the developer of NightOwl and from what he told me and based of your code snippet they both seem to use the same mechanic for changing the OS theme. Now I'll just have to figure out when I can reliably read the settings.

**Comment by weisJ** [comment]
@dobrovolsky @kuznecc If you change to "Auto" mode in the settings and then toggle to dark/light mode using NightOwl do the settings also get switched to dark/light or do they stay in "Auto" mode?

**Comment by kuznecc** [comment]
@weisJ The setting in the macOS properties remains unmodified by toggling the NightOwl, but the theme is changing. 
And it's no matter which setting is set up in the macOS properties. 
The appearance is changing, but the macOS setting isn't.

**Comment by dobrovolsky** [comment]
> @weisJ The setting in the macOS properties remains unmodified by toggling the NightOwl, but the theme is changing.
> And it's no matter which setting is set up in the macOS properties.
> The appearance is changing, but the macOS setting isn't.
It's changed. Try to change theme, close settings and open again.

@weisJ Same behaviour for auto mode, themes are swapped. It works fine only when you change theme manually from settings.
Will logs be useful for you?

**Comment by weisJ** [comment]
> @weisJ Same behaviour for auto mode, themes are swapped. It works fine only when you change theme manually from settings.

Do you mean Auto mode selected and the switching through NightOwl or Auto mode selected and waiting until the theme switches on it's own?

> Will logs be useful for you?

At the moment I don't log any relevant information on the native side (primarily because of the overhead involved with dispatching the log message to IntelliJ from the native layer). I'm currently working on a debug build with more detailed logging.

**Comment by weisJ** [comment]
What you could do for now though is run the following commands through the console after switching using NightOwnl
````
defaults read -g AppleInterfaceStyle
defaults read org.jetbrains.intellij NSRequiresAquaSystemAppearance
````
(I'm not sure if `org.jetbrains.intellij` is the correct identifier. You can find out the correct one using `osascript -e 'id of app "IntelliJ"'`)

Also please set your theme to light and then run the following:
````
osascript -e 'tell application "System Events"' -e 'tell appearance preferences' -e 'set dark mode to not dark mode' -e 'end tell' -e 'end tell'
defaults read -g AppleInterfaceStyle
````

**Comment by dobrovolsky** [comment]
Selecting auto mode and switching through NightOwl.
It is really strange case sometimes themes change correctly, sometimes doesn't change or change in opposite way.
```
❯ defaults read -g AppleInterfaceStyle
2020-08-14 17:51:29.866 defaults[99208:844080]
The domain/default pair of (kCFPreferencesAnyApplication, AppleInterfaceStyle) does not exist

~
❯ defaults read com.jetbrains.pycharm NSRequiresAquaSystemAppearance
0

~
❯ osascript -e 'tell application "System Events"' -e 'tell appearance preferences' -e 'set dark mode to not dark mode' -e 'end tell' -e 'end tell'

~
❯ defaults read -g AppleInterfaceStyle
Dark
```

**Comment by kuznecc** [comment]
@weisJ thanks for a workaround but honestly it's easier to switch the macOS theme with the OS system preferences instead of the NightOwl than to execute some console commands in advance to it.

**Comment by weisJ** [comment]
The command isn’t meant to be a workaround but to for me to see what the output is.

**Comment by kuznecc** [comment]
@weisJ Sorry, I miss understood
```
-- > before NightOwl
 ~ % defaults read -g AppleInterfaceStyle
Dark
 ~ % defaults read com.jetbrains.intellij NSRequiresAquaSystemAppearance
0
--> after nightOwl
 ~ % defaults read -g AppleInterfaceStyle
2020-08-14 22:01:33.167 defaults[24656:4031128]
The domain/default pair of (kCFPreferencesAnyApplication, AppleInterfaceStyle) does not exist
 ~ % defaults read com.jetbrains.intellij NSRequiresAquaSystemAppearance
0
```
 
Switched OS to the Light theme
```
--> before nightOwl
~ % defaults read -g AppleInterfaceStyle
2020-08-14 22:06:07.982 defaults[24879:4035748]
The domain/default pair of (kCFPreferencesAnyApplication, AppleInterfaceStyle) does not exist
~ % defaults read com.jetbrains.intellij NSRequiresAquaSystemAppearance
0
~ % -- after nightOwl
~ % defaults read -g AppleInterfaceStyle
Dark
 ~ % defaults read com.jetbrains.intellij NSRequiresAquaSystemAppearance
0
```

```
~ % osascript -e 'tell application "System Events"' -e 'tell appearance preferences' -e 'set dark mode to not dark mode' -e 'end tell' -e 'end tell'
defaults read -g AppleInterfaceStyle
Dark
```

**Comment by weisJ** [comment]
> @weisJ thanks for a workaround [...]

Does this mean using the command actually switches the IDE theme correctly?

**Comment by weisJ** [comment]
I have made some changes to how dark mode is detected which should make it more stable. @kuznecc @dobrovolsky could you please install the following version and check whether it works for you (Install from disk can be found by pressing the gear icon in the plugins tab)? Please uninstall any previous versions of the plugin and restart beforehand.

[auto-dark-mode-plugin-1.5.0-2020.2-night-owl-support.zip](https://github.com/weisJ/auto-dark-mode/files/5078952/auto-dark-mode-plugin-1.5.0-2020.2-night-owl-support.zip)

**Comment by kuznecc** [comment]
It's impossible to add the custom plugin version. The mac and base version fails to install, the auto-dark-mode-plugin-1.5.0-2020.2-night-owl-support.jar hangs my IDE and it's impossible to restart the IDE or start it after 'Force Quit'.
<img width="462" alt="Screenshot 2020-08-15 at 19 28 02" src="https://user-images.githubusercontent.com/2734183/90316819-7e143f80-df2d-11ea-9e54-27273ec5c675.png">

**Comment by weisJ** [comment]
You have to install the whole `.zip` file.

**Comment by kuznecc** [comment]
@weisJ Thanks. 
Successfully installed and this time the automatic theme switching works correctly.
You did an amazing job!

**Comment by dobrovolsky** [comment]
@weisJ Works great. Thank you!

**Comment by weisJ** [comment]
Great. I’ll release a new version with the changes as soon as possible.

**Comment by weisJ** [comment]
The fix is now available with version 1.5.1
